ForumProject/communications/consumers.py

```diff
@@ -0,0 +1,110 @@
+import json
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from .models import Room, Message
+
+
+class ChatConsumer(WebsocketConsumer):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(args, kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.room = None
+        self.messages = None
+        self.user = None
+        self.user_inbox = None
+
+    def connect(self):
+        self.room_name = self.scope['url_route']['kwargs']['room_name']
+        self.room_group_name = f'chat_{self.room_name}'
+        self.room = Room.objects.get(name=self.room_name)
+        self.messages = Message.objects.filter(room=self.room)
+        self.user = self.scope['user']
+        self.user_inbox = f'inbox_{self.user.first_name}'
+
+        self.accept()
+
+        async_to_sync(self.channel_layer.group_add)(
+            self.room_group_name,
+            self.channel_name,
+        )
+
+        self.send(json.dumps({
+            'type': 'user_list',
+            'users': [user.first_name for user in self.room.online.all()],
+        }))
+
+        if self.user.is_authenticated:
+            async_to_sync(self.channel_layer.group_add)(
+                self.user_inbox,
+                self.channel_name,
+            )
+            async_to_sync(self.channel_layer.group_send)(
+                self.room_group_name,
+                {
+                    'type': 'users_messages',
+                    'messages': [str(message) for message in self.messages],
+                }
+            )
```

Comment:
You send group messages (`users_messages`) right after adding the user to the group. Make sure the order of operations is correct: the user should be added to the group before broadcasting messages to ensure they receive all relevant communications.

Reply:
Thanks, I removed this code and resolved this issue by passing users messages from the view function to the template